content/blog/page-view-counter/index.md

```diff
@@ -0,0 +1,273 @@
+
+You can also access the [Azure Portal](https://portal.azure.com/), log in and see the resource group with the resources you just created!
+
+### Using Azure Storage
+
+Azure Storage encompasses many services such as Blobs, Tables, Queues, Files, etc. In our case, we mentioned we will use Tables because of the easy key-value storage. We can think of a page URL as the key, and the times it was counted as the value that we want to retrieve.
+
+At this point, we want to install the Azure Storage Tables NuGet package (or the equivalent for the language you want). We are actually going to use the [Cosmos DB Tables API](https://docs.microsoft.com/en-us/azure/cosmos-db/tutorial-develop-table-dotnet#install-the-required-nuget-package), since it includes the API for accessing storage tables.
+Let's add the reference to it:
+
+```csharp
+using Microsoft.Azure.Cosmos.Table;
+```
+
+Since we already created our account, we can easily access the Tables service through our code. What we need for this is the following:
+
+- The **connection string**: This is an authentication key that you can get [from the portal](https://docs.microsoft.com/en-us/azure/cosmos-db/tutorial-develop-table-dotnet#install-the-required-nuget-package).
+- A **Value** (view count) to store and retrieve in the tables through a **Key** (URL).
+- Code that connects and access the tables through the API package that we installed.
+
+Let us review how tables store entities work.
+
+### Table Entities
+
+The simple way to store entities is to create a class that inherits from `TableEntity`. What this means is that the class will need to have two Properties: A `PartitionKey` and a `RowKey` . Both keys, combined, form the **Key** through which we will **uniquely identify our entity**. However, separated they serve a different purpose:
+
+-  The `PartitionKey` defines the partition in which the service will store the entity: 
+>Tables are partitioned to support load balancing across storage nodes. A table's entities are organized by partition. A partition is a consecutive range of entities possessing the same partition key value. The partition key is a unique identifier for the partition within a given table, specified by the PartitionKey property. 
+
+-  The `RowKey` is a unique identifier for an entity within a given partition.
+
+  Since the `PartitionKey` is used for load balancing, it's important that it's not hardcoded. For this, we can use the URL of the page in which we are counting the views. The `RowKey` can be hardcoded, because we are already changing the key through the use of the URL in the `PartitionKey`. Eventually we could have different `RowKey` for different scenarios, but for now it will stay as a hardcoded value. You can read more [here](https://docs.microsoft.com/en-us/rest/api/storageservices/understanding-the-table-service-data-model).
```

Comment:
wouldn't the simplest scheme be to hardcode the `PartitionKey` and to make `rowKey` by the page we are trying to view count? Do we really need to load balance individual pages?

Reply:
In terms of simplicity, no, it would be the same.
In terms of performance, it would be decidedly worse because then it would all fall in the same partition. It's unlikely in this case that you would have very large amounts of data but the best practice is to never use the same partition for everything (in general, not only on Azure Storage).